image_search_engine/src/data/loader.py

```diff
@@ -0,0 +1,53 @@
+import torch
+import cv2
+import albumentations as A
+from image_search_engine.src import config as CFG
+
+class CLIPLoader(torch.utils.data.Dataset):
+    def __init__(self, image_filenames, captions, tokenizer, transforms):
+        """
+        image_filenames and cpations must have the same length; so, if there are
+        multiple captions for each image, the image_filenames must have repetitive
+        file names
+        """
+
+        self.image_filenames = image_filenames
+        self.captions = list(captions)
+        self.encoded_captions = tokenizer(
+            list(captions), padding=True, truncation=True, max_length=CFG.MAX_LENGTH
```

Comment:
max_length should be a parameter of the constructor. Let everything inside the constructor be parametrized and pass constants as arguments to the constructor.